### PR TITLE
Feature/d3 asim 2175

### PR DESCRIPTION
**Pull request by hannesdiedrich**

### CODE REVIEW

File: integration_tests/integration_tests.feature [hunk 1]
```diff
@@ -35,14 +35,12 @@ Feature: Run integration tests
...
 
-  # TODO: Once iaa_fee is included in billing, this test to be run non-zero iaa_fee
   Scenario Outline: Test Balanced Energy Bills
     Given we have a scenario named <scenario>
     And d3a is installed
-    When we run the simulation with setup file <scenario> and parameters [24, 60, 60, 0, 1]
+    When we run the simulation with setup file <scenario> and parameters [24, 60, 60, 1]
     Then the traded energy report the correct accumulated traded energy
     And the energy bills report the correct accumulated traded energy price
-    And the energy bills report the correct external traded energy and price
```
Comment (faizan2590) [inline integration_tests/integration_tests.feature]: Umm, can you explain the reason to remove this step?

File: integration_tests/grid_fee.feature [hunk 2]
```diff
@@ -109,7 +109,7 @@ Feature: GridFee integration tests
      And d3a uses an two-sided pay-as-clear market
      And the minimum offer age is <min_offer_age>
      And d3a dispatches events from top to bottom
-     When we run the simulation with setup file grid_fees.constant_grid_fees and parameters [24, 60, 60, 0, 1]
+     When we run the simulation with setup file grid_fees.constant_grid_fees and parameters [24, 60, 60, 1]
```
Comment (spyrostz) [inline integration_tests/grid_fee.feature]: Well, no harm having it here, but since most of the tests use the percentage fees, why not having a default value of percentage fees (without changing this parameter list) and if there is a need for a constant fee in a test, only then to add an extra parameter ? Since you did it this way it's not important now, just that you could have avoided the hassle of changing all the integration tests with this approach.

File: integration_tests/steps/integration_tests.py [hunk 3]
```diff
@@ -728,35 +709,6 @@ def test_accumulated_energy(context):
     assert isclose(net_energy, 0, abs_tol=1e-10)
 
 
-@then('the energy bills report the correct external traded energy and price')
```
Comment (spyrostz) [inline integration_tests/steps/integration_tests.py]: I think it is wrong to remove it now. Keeping the TODO there is important to not forget to reactivate it once the bug is resolved.

File: src/d3a/models/area/__init__.py [hunk 4]
```diff
@@ -43,14 +43,13 @@
 log = getLogger(__name__)
 
 
+# TODO: As this is only used in the unittests, please move to the tests package
```
Comment (spyrostz) [inline src/d3a/models/area/__init__.py]: I would say to delete this and replace the usages of this class with constants_limits.GlobalConfig class. If you can change the TODO to reflect that it would be great

File: integration_tests/grid_fee.feature (hunk 2 above)
Comment (hannesdiedrich) [reply integration_tests/grid_fee.feature]: Hmm i dont understand. There is no global grid_fee setting any more. The grid fees are set in the setup file. So i deleted this parameter. 
Or can you rephrase please?

File: integration_tests/integration_tests.feature (hunk 1 above)
Comment (hannesdiedrich) [reply integration_tests/integration_tests.feature]: Please have a look in to the step function , it is disabled anyway

File: integration_tests/steps/integration_tests.py (hunk 3 above)
Comment (hannesdiedrich) [reply integration_tests/steps/integration_tests.py]: IMO this test is totally outdated. There is no bug but another way of calculation. Shall I add another story to add  a real `the energy bills report the correct external traded energy and price` test?

File: src/d3a/models/area/__init__.py (hunk 4 above)
Comment (hannesdiedrich) [reply src/d3a/models/area/__init__.py]: Ok, will do!

File: integration_tests/grid_fee.feature (hunk 2 above)
Comment (spyrostz) [reply integration_tests/grid_fee.feature]: I see, I misread the parameters here. So the last parameter is always the market_count, right ? Then all is good.

File: integration_tests/alternative_pricing.feature [hunk 5]
```diff
@@ -9,7 +9,7 @@ Feature: Alternative Pricing
   Scenario Outline: Pricing for alternative pricing is right
     Given we have a scenario named <scenario>
     And d3a is installed
-    When we run the d3a simulation with <scenario> [24, 60, 60]
+    When we run the simulation with setup file <scenario> and parameters [24, 60, 60, 1]
```
Comment (spyrostz) [inline integration_tests/alternative_pricing.feature]: Regarding the latter comment, I originally was confused by this, because you added a parameter, and I incorrectly thought that it was the grid fee type. It is the market count though, and I assume you added the parameter here in order to remove a step definition, right ?

File: integration_tests/steps/integration_tests.py (hunk 3 above)
Comment (spyrostz) [reply integration_tests/steps/integration_tests.py]: Why not adding a link to the 2-sided PAC bug ticket? That way we know that this needs to be rewritten when the bug is resolved.

File: integration_tests/alternative_pricing.feature (hunk 5 above)
Comment (hannesdiedrich) [reply integration_tests/alternative_pricing.feature]: Exactly, this was bugging me for a while now, that we have two step functions for running a simple simulations.

File: integration_tests/steps/integration_tests.py (hunk 3 above)
Comment (hannesdiedrich) [reply integration_tests/steps/integration_tests.py]: Can you please provide the link to the ticket ? I cant find it.
So I leave the implementation here but link the ticket, yes?

Comment (hannesdiedrich) [reply integration_tests/steps/integration_tests.py]: (also disabling this step again, of course)

File: integration_tests/alternative_pricing.feature (hunk 5 above)
Comment (spyrostz) [reply integration_tests/alternative_pricing.feature]: Perfect, glad that this was removed.

File: integration_tests/steps/integration_tests.py (hunk 3 above)
Comment (spyrostz) [reply integration_tests/steps/integration_tests.py]: https://gridsingularity.atlassian.net/browse/D3ASIM-1887 , it is part of the TODO comment on the step implementation. I had a deeper look in the test, and I cannot understand why the test is outdated. I think it is perfectly valid still, and it essentially checks out if the trade_energy_price graph is consistent with the energy bills, as far as I understand. The only reason why this step should fail, is due to the behavior described in the bug. Why is it important to delete it actually ?

Comment (hannesdiedrich) [reply integration_tests/steps/integration_tests.py]: Ok, now it makes sense, thanks for charing the ticket!
Will re add the step again. Thanks!

File: integration_tests/integration_tests.feature (hunk 1 above)
Comment (hannesdiedrich) [reply integration_tests/integration_tests.feature]: Readding it again because of the other discussion
